openstates/de/bills.py

```diff
@@ -1,693 +1,339 @@
+                        vote_tds = vote_page.xpath(".//table//td")
+                        person_seen = False
+                        for td in vote_tds:
+                            if person_seen:
+                                person_vote = td.text_content().strip()
+                                if person_vote == "Y":
+                                    yes_votes.append(person)
+                                elif person_vote == "N":
+                                    no_votes.append(person)
+                                elif person_vote in ["NV","A","X"]:
+                                    other_votes.append(person)
+                                else:
+                                    raise AssertionError("Unknown vote '{}'".format(person_vote))
+                                person_seen = False
+                            else:
+                                person = td.text_content().strip()
+                                if person:
+                                    person_seen = True
+
+            if voice_vote:
+                vote = Vote(vote_chamber,date,"passage",passed,0,0,0)
+            else:
+                vote = Vote(vote_chamber,date,"passage",
+                            passed,yes_count,no_count,other_count,
+                            yes_votes=[],
+                            no_votes=[],
+                            other_votes=[])
+
+                vote["yes_votes"] = yes_votes
+                vote["no_votes"] = no_votes
+                vote["other_votes"] = other_votes
+
+            assert len(vote["yes_votes"]) == vote["yes_count"], \
+                "Yes vote count does not match number of yes votes"
+            assert len(vote["no_votes"]) == vote["no_count"], \
+                "No vote count does not match number of no votes"
+            assert len(vote["other_votes"]) == vote["other_count"], \
+                "Other vote count does not match number of other votes"
+
+            if (passed
+                and vote["yes_count"] <= vote["no_count"]
+                and not voice_vote):
+                    raise AssertionError("Vote passed with more N than Y votes?")
+
+            if not passed and vote["yes_count"] > vote["no_count"]:
+                self.logger.warning("Vote did not pass but had a majority \
+                        probably worth checking")
+
+            vote.add_source(doc)
+            bill.add_vote(vote)
+
+        bill.add_source(link)
+
+        return bill
+
+
+    def scrape(self,chamber,session):
+        bill_codes = {"lower":[1,2,3,4],"upper":[5,6,7,8]}
+        base_url = "http://legis.delaware.gov/LIS/lis{session}.nsf/ByAll?OpenForm&Start=1&Count={count}&Expand={bill_type}&Seq=1"
+        for bill_code in bill_codes[chamber]:
+            page = self.lxmlize(base_url.format(session=session,
+                                            bill_type=bill_code,
+                                            count=1000)) #1000 shld be enough
```

Comment:
Maybe note that previous sessions in the Open States database haven't had more than 1,000 bills _total_.